lib/tss/response_handlers.rb

```diff
@@ -16,4 +16,12 @@ def handle_get_response(response, path, root_name)
     else fail "TSS: 'GET #{path}' failed with status = #{response.code}."
     end
   end
+
+  def handle_put_response(response, path)
+    case response.code
+    when 200 then JSON.parse(response.body)
+    when 404 then nil
+    else fail "TSS: 'PUT #{path}' failed with status = #{response.code}."
+    end
+  end
```

Comment:
sounds good
> On Feb 18, 2016, at 11:19 AM, Tamás Majoros <notifications@github.com> wrote:
> 
> In lib/tss/response_handlers.rb <https://github.com/shore-gmbh/payment-connector/pull/11#discussion_r53295160>:
> 
> > @@ -16,4 +16,12 @@ def handle_get_response(response, path, root_name)
> >      else fail "TSS: 'GET #{path}' failed with status = #{response.code}."
> >      end
> >    end
> > +
> > +  def handle_put_response(response, path)
> > +    case response.code
> > +    when 200 then JSON.parse(response.body)
> > +    when 404 then nil
> > +    else fail "TSS: 'PUT #{path}' failed with status = #{response.code}."
> > +    end
> > +  end
> Thanks, it looks good, I'll try to use this one (we also have handle the nil response in case of 404 error according to the specs)
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/shore-gmbh/payment-connector/pull/11/files#r53295160>.
> 


-- 



www.shore.com


*Besuchen Sie uns auch auf  *Google+ 
<https://plus.google.com/u/0/111422551090494373794/posts>
Facebook <http://www.facebook.com/shore>
Shore Blog <http://blog.shore.com/>


*Shore GmbH *Rosenheimer Straße 145e
81671 München
Germany


Eingetragen beim Amtsgericht München, HRB 196674 
Geschäftsführer: Alexander Henn, Philip Magoulas